### ISSUE TITLE
Slow gain changes (> 3 dB!) on Booster 003B002E3037470535353239

### ISSUE THREAD

**Issue by dnadlinger** [sinara-hw/Booster#329]
## Observation

I am seeing the gain of one channel change by >3 dB on a TechnoSystem v1.4 Booster (and potentially also a few other Creotech/TechnoSystem devices, although without good data).

The Booster in question is usually driven by an Urukul and drives an AOM. Measuring the powers directly with an RF power meter at two points, when the gain appears high/low, gave:

 * "High gain": -11.45 dBm input at 210 MHz gives 28.29 dBm output -> 39.74 dB gain
 * "Low gain": -11.48 dBm input at 210 MHz gives 24.42 dBm output -> 35.90 dB gain

There was no clear short-term trend visible, but the gain had changed back and forth between roughly these values over a few days.

## Debugging information

The problem occurs on channel #0 on Booster 003B002E3037470535353239, which is a factory-refurbished older release.

No I2C errors, nothing interesting in the log:

```
> i2cerr
		#0	#1	#2	#3	#4	#5	#6	#7
I2C ERR		0	0	0	0	0	0	0	0
```

```
> logstash
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[INFO] network client disconnected
[INFO] network client disconnected
[INFO] network client 10.255.6.191 connected
[ERROR] Interlock tripped on channel 1, i=0 o=1
```
(Channel 1 is unrelated.)

Status without RF applied:
```
PGOOD: 1
FAN SPEED: 20 %
CHANNELS INFO
==============================================================================
		#0	#1	#2	#3	#4	#5	#6	#7
DETECTED	1	1	1	1	1	1	1	1
HWID		02:22	CD:F9	7B:9F	20:8D	E8:27	4F:79	21:12	C6:3A
INPWR [V]	0.00	0.13	0.00	0.08	0.02	0.01	0.00	0.05
TXPWR [V]	0.01	0.01	0.15	0.12	0.01	0.01	0.01	0.02
RFLPWR [V]	0.04	0.01	0.01	0.01	0.01	0.22	0.13	0.08
INPWR [dBm]	nan	nan	nan	nan	nan	nan	nan	nan
TXPWR [dBm]	5.00	5.00	5.00	5.00	5.00	5.00	5.00	5.00
RFLPWR [dBm]	-4.31	-4.48	-4.24	-3.55	-2.58	-2.93	-0.01	-4.04
I30V [A]	0.028	0.042	0.048	0.046	0.042	0.049	0.051	0.045
I5V0 [A]	0.246	0.255	0.245	0.249	0.250	0.249	0.253	0.256
5V0MP [V]	4.934	4.930	4.940	4.950	4.920	4.942	4.940	4.962
ON		1	1	1	1	1	1	1	1
SON		1	1	1	1	1	1	1	1
IINT		0	0	0	0	0	0	0	0
OINT		0	0	0	0	0	0	0	0
SINT		0	0	0	0	0	0	0	0
ADC1		15	19	244	195	13	13	13	38
ADC2		63	18	15	13	13	364	205	123
INTSET [dBm]	31.04	30.84	30.35	28.99	36.87	36.67	35.49	33.59
DAC1		4095	4095	4095	4095	4095	4095	4095	4095
DAC2		3413	3424	3527	3341	3992	3896	3924	3821
SCALE1		83	85	82	83	87	88	85	87
OFFSET1		470	446	727	619	460	375	565	571
BIASCAL		1865	1539	1879	1935	1527	1939	1761	1929
HWIS		82.08	84.33	83.17	82.83	85.17	83.92	83.00	85.25
HWIO		865.08	823.33	1003.17	939.83	852.17	818.92	978.00	957.25
LTEMP		27.50	28.25	28.50	27.00	28.50	27.00	27.00	28.00
RTEMP		27.50	27.00	27.00	28.00	27.00	28.25	28.25	27.00
==============================================================================
```

Status with above RF power applied:
```
PGOOD: 1
FAN SPEED: 20 %
CHANNELS INFO
==============================================================================
		#0	#1	#2	#3	#4	#5	#6	#7
DETECTED	1	1	1	1	1	1	1	1
HWID		02:22	CD:F9	7B:9F	20:8D	E8:27	4F:79	21:12	C6:3A
INPWR [V]	0.80	0.89	0.00	0.08	0.03	0.01	0.00	0.05
TXPWR [V]	1.57	1.81	> 0.15	0.12	0.01	0.01	0.01	0.03
RFLPWR [V]	0.64	1.06	0.01	0.01	0.01	0.22	0.13	0.07
INPWR [dBm]	nan	nan	nan	nan	nan	nan	nan	nan
TXPWR [dBm]	25.41	29.72	5.00	5.00	5.00	5.00	5.00	5.00
RFLPWR [dBm]	8.01	16.78	-4.24	-3.54	-2.58	-2.95	-0.01	-4.05
I30V [A]	0.090	0.162	0.048	0.046	0.042	0.049	0.051	0.045
I5V0 [A]	0.244	0.254	0.245	0.249	0.250	0.249	0.253	0.257
5V0MP [V]	4.930	4.928	4.940	4.950	4.918	4.942	4.942	4.962
ON		1	1	1	1	1	1	1	1
SON		1	1	1	1	1	1	1	1
IINT		0	0	0	0	0	0	0	0
OINT		0	0	0	0	0	0	0	0
SINT		0	0	0	0	0	0	0	0
ADC1		2578	2972	245	196	14	14	13	42
ADC2		1051	1741	15	14	14	366	213	121
INTSET [dBm]	31.04	30.84	30.35	28.99	36.87	36.67	35.49	33.59
DAC1		4095	4095	4095	4095	4095	4095	4095	4095
DAC2		3413	3424	3527	3341	3992	3896	3924	3821
SCALE1		83	85	82	83	87	88	85	87
OFFSET1		470	446	727	619	460	375	565	571
BIASCAL		1865	1539	1879	1935	1527	1939	1761	1929
HWIS		82.08	84.33	83.17	82.83	85.17	83.92	83.00	85.25
HWIO		865.08	823.33	1003.17	939.83	852.17	818.92	978.00	957.25
LTEMP		27.50	28.25	28.50	27.00	28.50	27.00	27.00	28.00
RTEMP		27.50	27.00	27.00	28.00	27.00	28.25	28.25	27.00
==============================================================================
```

Note the low bias current.

Disabling and re-enabling the channel using the console (`disable 1; enable 1`) doesn't change much about the gain.

Bias currents after successively disabling and re-enabling channel 0: 32 mA, 37 mA ,41 mA, 43 mA, still 43 mA, 44 mA, 45 mA. Under load, still only 88 mA and 25.25 dBm forward power, though (using internal detector, but seems to roughly match power meter).

(Lack of) gain is stable across power-cycling entire Booster. Initially had 36 mA current after startup, settled to 41 mA after a few minutes.

Channel 0 supply current vs. input RF amplitude (in Urukul full-scale units, where 1.0 corresponds to the above measurements):

| DDS amplitude / FS | I30V / A |
|--------------------|----------|
| 0.0 | 0.032 |
| 0.1 | 0.038 |
| 0.2 | 0.042 |
| 0.3 | 0.049 |
| 0.4 | 0.057 |
| 0.5 | 0.059 |
| 0.6 | 0.064 |
| 0.7 | 0.069 |
| 0.8 | 0.074 |
| 0.9 | 0.081 |
| 1.0 | 0.089 |

## Extra observations

While attempting to gather data on this Urukul -> Booster -> AOM channel behaviour a few days ago, I saw the gain initially starting out low when switching on the Urukul channel. About 0.5 s later, the output power would recover to the high value in what looked very much like a step change. (This was done while acquiring AOM optical output power readings, i.e. by proxy RF power input measurements, for points between zero and full RF amplitude in random order. The gain really did appear to change for all input powers.)  I couldn't conclusively establish this as being caused by Booster, as the problem went away after some 10 minutes before I could take a closer look.

**Comment by dnadlinger** [comment]
(I'm not sure what is going on here. Some intermittent almost-short? Weird damage to the power FET or biasing circuitry? Other RF black magic seems unlikely, as the bias current is low, not high, and it occurs without input.)

**Comment by hartytp** [comment]
Thanks for the report.

I plan to post this booster back to creotech tomorrow for investigation (@dnadlinger does it also have watchdog issues?)

**Comment by hartytp** [comment]
We dismissed these problems as damaged FETs before, but I don’t see how the FETs could become damaged when there is fold back limiting.

Can’t be rf oscillations as that would push the bias current up normally not down

Can’t be disconnected bias voltage as that would push the current up

The fold back protection hasn’t tripped as the current can still increase with input rf power

Quite mysterious...

**Comment by gkasprow** [comment]
I hope to be able to recreate it in my lab. This is really mistery. I will connect the channel outside of the booster to exclude the influence of the other circuits.

**Comment by hartytp** [comment]
Yes, I’ll definitely owe you a beer if you can figure this one out!

**Comment by hartytp** [comment]
FYI I if you look at they hysteresis issue you’ll see that I found that when I power cycled the booster the 28V voltage and gate bias current were reproducible to better than 1 mag. However the bias current (measured directly across the sense resistor ) could change by 10% or more

**Comment by gkasprow** [comment]
We used to have some hysteresis a long time ago when the power stage output indictor had too low value. I could not explain why this happened, but since higher value solved the issue, we didn't dig any deeper.

**Comment by gkasprow** [comment]
Could be also an issue with the output capacitor. Some ceramic caps behave in a bizarre way when high current is passed at higher frequencies. Changing to bigger package size helps. I already experienced such a situation in my amplifier working with 10W and 400MHz a few weeks ago.

**Comment by hartytp** [comment]
Yes but that doesn’t explain this since it happens at dc with no rf signal...

**Comment by dnadlinger** [comment]
Also, the ("good") output power I'm using is only ~ 29 dBm, so if it was just related to raw power, surely it would routinely occur on channels tested on full blast?

**Comment by hartytp** [comment]
@gkasprow we're shipping two Boosters to Creotech today.

One has a missing channel (the one with a short we sent back). At some point the watchdog tripped on this booster and now it won't start up properly even after power cycling.

The other one has all channels populated, and is the one with the gain variations. I believe that both should manifest the watchdog issues.

**Comment by gkasprow** [comment]
I managed to recreate the problem with one channel you sent us previously.
This channel shows rapid gain changes. It does it from time to time, mostly during a few minutes after power on. Later on, it seems to work correctly.
I'm observing the signal at the power stage entry and it is stable.
I plan to look at the gate DC voltage and current using same scope.

**Comment by hartytp** [comment]
FWIW it's not clear that this is really an issue with the power stage. Could be anywhere in the signal chain. So the first thing to do is probably to trace the signal through Booster and see where the issue really is.

**Comment by gkasprow** [comment]
As I wrote, signal at the gate looks OK, the same with its DC level. I will look at the current

**Comment by hartytp** [comment]
aah, ok, sorry I mis-understood you. So...it's definitely the power FET/matching then.

**Comment by gkasprow** [comment]
it looks so. However it seems that after heating up it works.

What is the issue with the channel you posted me a few days ago?

**Comment by hartytp** [comment]
>What is the issue with the channel you posted me a few days ago? 

https://github.com/sinara-hw/Booster/issues/332

**Comment by hartytp** [comment]
@gkasprow which channel were you looking at? Did you read @dnadlinger's post at the top of this issue? In channel 0 (ID 02:22) the gain is unstable and slowly varies between high and low values. There is a corresponding change in the bias current.

The bias current also shows significant hysteresis between power cycles (but, interestingly not between enabling/disabling the channels). I've observed this behaviour on other channels, but not as bad.

**Comment by hartytp** [comment]
The bias current for this channel is generally lower than the 50mA it was calibrated for.

**Comment by hartytp** [comment]
> However it seems that after heating up it works.

We observed the gain issues to come and go.

**Comment by dnadlinger** [comment]
> However it seems that after heating up it works.

[…]

>  We observed the gain issues to come and go.

Just to reiterate, we definitely observed large gain changes with the module, i.e. the entire Booster, constantly powered on.

(This is talking about bias power, though; the RF signal on that channel does vary as part of the experiments in the ion trap, between completely off, different power levels, and a few tens of MHz of frequency difference.)

**Comment by gkasprow** [comment]
the issue is more complex
the output transistor seems to ge internal avalanche discharge.
Look at the plots:
![tek00002](https://user-images.githubusercontent.com/4325054/70074933-05b65a00-15fc-11ea-9219-ec778701b11e.png)
![tek00003](https://user-images.githubusercontent.com/4325054/70074940-07801d80-15fc-11ea-93c1-dead96b7360d.png)
![tek00004](https://user-images.githubusercontent.com/4325054/70074944-0949e100-15fc-11ea-8e38-8859ee0f7f7d.png)

**Comment by gkasprow** [comment]
interesting plot
![tek00008](https://user-images.githubusercontent.com/4325054/70075472-2d59f200-15fd-11ea-967d-b0dfa2c0652a.png)
![tek00009](https://user-images.githubusercontent.com/4325054/70075480-30ed7900-15fd-11ea-8df7-c7e725580b0e.png)

**Comment by gkasprow** [comment]
the voltage on gate looks very distorted. And the power stage seems to amplify correctly.

**Comment by gkasprow** [comment]
It looks like the first stage got crazy

**Comment by hartytp** [comment]
Aah amazing! So we’re closing in on the real issue now...

**Comment by gkasprow** [comment]
I replaced the amp and now it works better.
No idea why it behave like this. It could be damaged by excessive input power.

**Comment by hartytp** [comment]
Hmm...

Which amp did you replace? How can it be damaged when we have the interlock?

I think the input stage playing up could be the cause of many issues (intermittent noise/hysteresis on the 29V bias current, strange variations etc) which I have seen on many channels including some that are fresh from the supplier. So I think there is a systematic issue here

**Comment by hartytp** [comment]
We need to get to the bottom of this and not just brush it under the carpet

**Comment by gkasprow** [comment]
I replaced the first stage only. it is supplied from 5V. It could be damaged during experimental work. I don't know history of this channel.
However, after replacement, I've noticed that the gain drifts slowly after power-up. So there is still something happening here. I cannot re-create it anymore. Could be also cool-down effect after amp soldering but I doubt.

**Comment by gkasprow** [comment]
I cycled the power and the gain is stable. But I'm pretty sure I saw a slow drift by something like 3dB...

**Comment by gkasprow** [comment]
I will continue tomorrow. It's time to go home.

**Comment by gkasprow** [comment]
I left it running with full power overnight

**Comment by gkasprow** [comment]
I'm thinking about the possible causes of ADL5536 amplifier damage.
It is loaded with a high pass filter which has a very low capacitance value of 39pF. On the other had it has 1uH load. What if we overdrive it? The output stage is just a common emitter transistor that switches between saturation and off state. And what happens with inductor energy when the transistor switches to off state? It generates a high voltage that damages the transistor. Its energy is not that high to damage it thermally, it just causes avalanche discharge and degradation of the silicon structure. Such a situation could happen here. Solution could be simple - just add low capacitance TVS (or two silicon diodes in series) or move the highpass filter after the attenuator.
I will try to recreate such a condition tomorrow.
Here is the simulation I did. I used a popular RF transistor and a similar LC circuit.
![obraz](https://user-images.githubusercontent.com/4325054/70090528-a9166780-161a-11ea-8c96-6fd7e59229e6.png)

**Comment by gkasprow** [comment]
the reference design uses 100nF cap and 50Ohm load of next stage so  we get much lower overshoot
![obraz](https://user-images.githubusercontent.com/4325054/70090926-66a15a80-161b-11ea-8f76-c1053926e2e3.png)

**Comment by gkasprow** [comment]
We can cause such conditions for example when activating the input interlock or during disconnecting or connecting the input cable. Anyway, I will try to damage the amplifier I replaced in such a way tomorrow observing the output with a low capacitance active probe.

**Comment by gkasprow** [comment]
The output stage has a max voltage of 100V so it gives us safe margin.
![obraz](https://user-images.githubusercontent.com/4325054/70093286-593a9f00-1620-11ea-8f6a-2e519758bc4f.png)

**Comment by gkasprow** [comment]
The second stage
![obraz](https://user-images.githubusercontent.com/4325054/70093509-d9610480-1620-11ea-811c-d3b32d8ac292.png)

**Comment by hartytp** [comment]
@gkasprow what are you measuring in these scope traces?

**Comment by gkasprow** [comment]
Green is amplifier output via 20dB attenuator
Yellow is power stage gate

**Comment by hartytp** [comment]
With RF input applied? Or, are you saying the amplifier was free oscillating at around 400MHz with input terminated? RMS around 1.2V so about 14dBm?

**Comment by hartytp** [comment]
Also, is the power stage really amplifying correctly? The green curve doesn't look like just a scaled up version of the yellow curve.

**Comment by gkasprow** [comment]
I used a 400MHz generator all the time. -10dBm
The amplifier was itself switching between some internal states :)

**Comment by gkasprow** [comment]
It was caused by broken first stage which sometimes was amplifying, sometimes was terribly distorting the input signal.

**Comment by hartytp** [comment]
I see. So the observation is that the gate voltage is distorted and the output switches between different gains as previously observed. OK.

**Comment by gkasprow** [comment]
not really. The output simply follows the gate voltage. But I observe the output after the HP filter so it does not look that bad as the input.

**Comment by hartytp** [comment]
Without any signal applied did you see any AC at the PA FET's gate/output?

**Comment by gkasprow** [comment]
nope.

**Comment by gkasprow** [comment]
It looks like the gain changes were caused by imput amplifier. I replaced it with one disassembled from broken channel. So it can be broken as well.

**Comment by hartytp** [comment]
Okay. I suspect that there is more than one issue here. The symptoms we have observed are:
1. gain changes
2. sometimes the gain changes are accompanied by a change in the PA FET DC bias (i.e. the current is lower even with no RF applied)
3. Hysteresis in the FET bias current when the unit is turned on/off (going in/out of standby doesn't affect this. It's also not thermal since the current seems stable until the unit is power cycled again)
4. Intermittent periods when the 29V FET bias current becomes nosier

(1) could be explained by damaged first stage amps
(2) can't be explained by damaged first stage amps
(3)-(4) could be explained if the first stage amps were oscillating or something like that.

Anyway, if the first stage amp is damaged we should definitely look at that first, but it may be that there is more than one issue here...

**Comment by hartytp** [comment]
NB lower gains correspond to lower DC bias currents. That suggests there is something wrong in the FET/biassing circuit. However, last time I looked the bias voltage on the supplemental control voltage wasn't changing.

**Comment by hartytp** [comment]
@gkasprow how do you know it was the ADL and not the PHA? Did you trace the signal through the circuit and see where the distortion came in?

**Comment by gkasprow** [comment]
yes. I used <1pF oscilloscope probe. To access the lower board I have a ritht angled goldin adapter and pair of DSUB connectors connected with 10cm wires.

**Comment by hartytp** [comment]
> yes.

"Yes", as in you traced the signal through and confirmed that the first amp was causing the distortion?

If so, that's great! We've caught one place where there is damage so we can fix that...

**Comment by gkasprow** [comment]
exactly.

**Comment by hartytp** [comment]
NB the ADL can take up to 20dBm input power. Given that there is approx 7dB of loss before the amp it should take nearly 0.5W of input power to damage it. The only thing we've used to drive Booster is Urukul so I think we can safely say that input overdrive is not the issue here!

**Comment by gkasprow** [comment]
Yes, but it can lead to output overvoltage.

**Comment by hartytp** [comment]
I don't see a spec on max collector voltage.

**Comment by gkasprow** [comment]
it's probably the same as twice the supply voltage

**Comment by hartytp** [comment]
Is that an RF rule of thumb? So the worry would be some kind of LC resonance in the output stage? Maybe coupled with a switch glitch either from Urukul or from the interlock?

**Comment by gkasprow** [comment]
I assume the sine wave and AC coupled load. Output can go to zero and twice the supply. The max power supply is then transistor breakdown value /2. I can measure the breakdown voltage tomorrow.

**Comment by hartytp** [comment]
I guess the other worry is if the RF is somehow applied before the amp is properly biassed?

**Comment by gkasprow** [comment]
without biasing there is no overvoltage induced.

**Comment by hartytp** [comment]
No, but isn't there a possible issue of damaging the input stage with RF applied before bias?

**Comment by dnadlinger** [comment]
@hartytp: I guess at abs max. V_cc you'd assume (without further information) that a symmetric signal from 0 to 2 V_cc is the maximum permissible.

@gkasprow: Thanks a lot for concentrating on these issues now – really looking forward to when Booster just works!

**Comment by hartytp** [comment]
Okay, so anything above ~12V could cause breakdown?

**Comment by gkasprow** [comment]
the IC is just Darlington transistor with matched input

**Comment by gkasprow** [comment]
exactly

**Comment by hartytp** [comment]
Okay. That makes sense. Could be what is killing the input stage. As you say, it's worth measuring the spikes we get when the interlock trips (maybe also hook up to Urukul and look at those transients) and see how it compares.

**Comment by gkasprow** [comment]
RF amplifiers usually have a similar schematic
![obraz](https://user-images.githubusercontent.com/4325054/70096501-8474bc80-1627-11ea-9001-747006d76459.png)

**Comment by hartytp** [comment]
In an earlier version the input attenuator was after the initial pre-amp stage.

We moved it to give a better input return loss, but that's really not necessary IMHO. If there is an issue, we could always move the input attenuator to after the pre-amp. That would isolate it from the load.

**Comment by gkasprow** [comment]
Tomorrow I plan:
- measure amplifier breakdown voltage
- measure spikes it produces during interlock activation or applying square wave signal
- looking deeper to slow gain changes that I observed today
- looking deeper into the hysteresis issue.

**Comment by hartytp** [comment]
Wonderful! Thanks!

**Comment by hartytp** [comment]
@gkasprow I think your plan for tomorrow sounds good.

AFAICT we've got damage to both the power FET and the first pre-amp in different boards. So, we should check the whole design by looking for voltage/bias current transients during: turn on/off; interlock trip/reset; standby/enable; Urukul switch transients? Basically check the whole thing over for any transient issue that can take one of the amp ICs outside its abs max ratings.

**Comment by gkasprow** [comment]
The channel I left working with full power overnight died. The first stage does not pass the signal anymore. Yesterday I assembled used chip which had some issues.

**Comment by gkasprow** [comment]
The amplifier died in a strange way - the input bias is 0

**Comment by hartytp** [comment]
the pre-amp input bias is 0V???

**Comment by gkasprow** [comment]
yes

**Comment by gkasprow** [comment]
but as I said, the chip was taken from a broken amplifier.

**Comment by hartytp** [comment]
what do the diagnostics (e.g. VCP `start`) say?

Anyway, at least we've identified on issue: the pre-amps are failing. That's a good start.

**Comment by gkasprow** [comment]
It's connected to slot 4
PGOOD: 1
FAN SPEED: 0 %
CHANNELS INFO
==============================================================================
		#0	#1	#2	#3	#4	#5	#6	#7
DETECTED	0	1	1	1	1	1	1	1	
HWID		00:00	B1:CF	90:A2	69:E0	E9:4F	56:81	39:6B	AE:20	
INPWR [V]	0.00	0.18	0.56	0.00	1.62	0.20	0.00	0.18	
TXPWR [V]	0.79	0.04	0.01	0.01	0.46	0.01	0.01	0.11	
RFLPWR [V]	0.81	0.01	0.01	0.01	0.01	0.06	0.01	0.01	
INPWR [dBm]	0.00	nan	nan	nan	nan	nan	nan	-19.00	
TXPWR [dBm]	0.00	5.00	5.00	5.00	5.00	5.00	5.00	5.00	
RFLPWR [dBm]	0.00	-2.63	-3.52	-4.30	-3.18	-4.07	-5.70	-3.68	
I30V [A]	0.000	0.050	0.050	0.055	0.032	0.051	0.059	0.043	
I5V0 [A]	0.000	0.254	0.281	0.265	0.142	0.260	0.264	0.261	
5V0MP [V]	0.000	4.980	4.971	4.973	4.965	4.966	4.953	4.976	
ON		0	1	1	1	1	1	1	1	
SON		0	1	1	1	1	1	1	1	
IINT		0	0	0	0	0	0	0	0	
OINT		0	0	0	0	0	0	0	0	
SINT		0	0	0	0	0	0	0	0	
ADC1		1293	59	12	12	755	23	12	181	
ADC2		1319	11	12	11	15	106	12	11	
INTSET [dBm]	0.00	30.00	30.49	30.00	35.00	33.00	31.50	33.68	
DAC1		0	4095	4095	4095	4095	4095	4095	4095	
DAC2		0	3444	3328	3259	3385	3600	3451	3825	
SCALE1		0	87	87	85	77	87	87	83	
OFFSET1		0	693	562	600	590	635	594	664	
BIASCAL		0	1960	1690	1700	1299	1687	1756	1651	
HWIS		0.00	90.00	89.00	87.00	82.00	87.00	88.00	84.33	
HWIO		0.00	744.00	614.00	649.00	515.00	729.00	679.00	984.33	
LTEMP		0.00	31.00	34.50	33.00	36.00	34.00	34.50	32.00	
RTEMP		0.00	32.25	32.00	34.00	37.25	34.25	33.00	33.00	
==============================================================================

**Comment by gkasprow** [comment]
Another issue is too low current of the power stage

**Comment by hartytp** [comment]
Okay, so in this failure mode the 5V0 bias current is noticably lower than normal. I don't believe I've ever observed that, so it's a new failure mode. Joy of joys :)

The low PA current is something I have observed before. I wonder if the answer here is that voltage spikes propagate through the system and trash the three amps at various levels, with the FETs being the most sensitive?

**Comment by gkasprow** [comment]
I did some simulation of the FET stage but did not manage to exceed 100V on its output. But Let's measure it.

**Comment by hartytp** [comment]
Do you think it's excessive gate voltage when a high-frequency spike is produced?

**Comment by hartytp** [comment]
Otherwise, I have no idea how they are still dying so often

**Comment by gkasprow** [comment]
ah, yes. this could be an issue.

**Comment by gkasprow** [comment]
we can easily exceed max ratings for the power stage
![obraz](https://user-images.githubusercontent.com/4325054/70137427-29c27b80-168e-11ea-895a-2f88830301d9.png)

**Comment by hartytp** [comment]
Well, the right approach here is to verify that we exceed that and ideally see a change in FET performance once we do. If that happens we can be decently confident we've identified the issue. (I had checked that max rating, but thought we were comfortably inside it because I hadn't appreciated how bad the ringing on the output stage can be...)

**Comment by gkasprow** [comment]
Broken gate has increased leakage current

**Comment by hartytp** [comment]
> Broken gate has increased leakage current

How would the gate become damaged? have you observed a voltage spike there? Have you measured an increased leakage current?

Last time I checked, the bias voltage measured on the supplemental PCB seemed very stable. Admittedly that's only got a 4R7 impedance, while there is another 39R on the PA PCB. But if the gate leakage were changing a lot I would have expected to see changes in OUT_BIAS measured on the supplemental PCB. Unless something odd is happening like the feedback coupling large output spikes onto the input and damaging it that way?

**Comment by hartytp** [comment]
@gkasprow any more progress on this?

**Comment by gkasprow** [comment]
Gate can be damaged with too high voltage. It causes leakage which can be measured using ohmmeter. Yuo may not see it with current reading

**Comment by hartytp** [comment]
Okay, can you check for this.

**Comment by gkasprow** [comment]
I will. I have unexpected paperwork today.

**Comment by hartytp** [comment]
Understood. Hopefully we can finish the testing this week. I'm keen to get some repaired/patched units asap so we can start gathering statistics and see if all issues are finally fixed.

**Comment by gkasprow** [comment]
I'm back in the lab

**Comment by hartytp** [comment]
:)

**Comment by gkasprow** [comment]
I took new non-damaged unit. I was looking at the first stage output with 3.9pF scope probe. I applied 1V 1kHz square signal to the input. The positive pulses at the stage output do not exceed 10V (5V above 5V supply)

**Comment by gkasprow** [comment]
however this unit has other L2 model. It looks like FB, not wire-wound like the one in faulty channel. I will replace the ADL chip with another one in the PCB that caused troubles.

**Comment by hartytp** [comment]
how large are the switch transients?

**Comment by gkasprow** [comment]
these transients do not trigger interlock
![tek00011](https://user-images.githubusercontent.com/4325054/70170001-306fe380-16cc-11ea-98fb-a8a6ab156b29.png)
yellow - output of the first stage
green - channel output after 20dB attenuator

**Comment by gkasprow** [comment]
![tek00014](https://user-images.githubusercontent.com/4325054/70170805-b9d3e580-16cd-11ea-9641-bc327be9ae2d.png)

interlock activation
blue - 1st stage output
yellow - input switch control signal

**Comment by gkasprow** [comment]
![tek00018](https://user-images.githubusercontent.com/4325054/70171163-729a2480-16ce-11ea-973a-8836219fe623.png)
yellow - 1st stage output
green - channel output
I applied 1.5Vpp 10Hz square signal to the input

**Comment by gkasprow** [comment]
I looked at the power stage gate in same conditions
green - PA output
yellow - power stage gate
rising edge and folling edge
![tek00020](https://user-images.githubusercontent.com/4325054/70171487-17b4fd00-16cf-11ea-91e6-1e8a46a0ca32.png)
![tek00019](https://user-images.githubusercontent.com/4325054/70171496-1d124780-16cf-11ea-98e2-672a71f79651.png)

**Comment by gkasprow** [comment]
the negative gate voltage is -4.5V. It exceeds max rating by 1.5V

**Comment by gkasprow** [comment]
here one can see interlock activation - violet trace
![tek00021](https://user-images.githubusercontent.com/4325054/70172116-58f9dc80-16d0-11ea-9761-e8503a62517b.png)

**Comment by gkasprow** [comment]
edit. the negative gate voltage is 10V, so still within the limit. The same as positive one

**Comment by hartytp** [comment]
hmmm...so no evidence of anything that could cause damage yet...

**Comment by gkasprow** [comment]
I'm curious why the FET output has such strange shape.

**Comment by hartytp** [comment]
Thanks Greg!

**Comment by gkasprow** [comment]
It's quite possible that I'm working with already partially damaged amplifiers. There is a chance that one can get broken after the first power-on cycle.
I'll order a bunch of new ones and replace them.
I run out of ADL ones while still have the other ones in my lab.

**Comment by hartytp** [comment]
Maybe. Something is happening that systematically damages these amps. I’ve never seen anything like it

**Comment by hartytp** [comment]
Before we get too convinced the amps are dying are we sure it’s not something like: mechanical issue creating a high Z intermittent short; problem with the bias; sequencing issue (eg input applied before bias); problemmwith matching circuit element; etc etc

**Comment by hartytp** [comment]
If we take a “damaged” component out and put it on an eval board/deadbug does it still not work?

**Comment by gkasprow** [comment]
I replaced the amp that died yesterday and left the booster working with full power over the night with scope connected and triggered by the output signal. If it dies we have a chance to catch the cause.
The dead chip has 5V at its output and 0 at its input. I didn't throw it away so I can check it tomorrow.
Today I had plans to do a lot of work but unexpected paperwork appeared. The report deadline was today :)

**Comment by hartytp** [comment]
Ok. Hopefully tomorrow is our day. Is there anyone else at wut/cti who could lend a fresh out of eyes?

**Comment by gkasprow** [comment]
Once I get new ICs, I will add 12V TVS and will see where the clamping occurs.

**Comment by gkasprow** [comment]
My colleague, Wojtek, was looking at it today and proposed a few ideas to study.

**Comment by gkasprow** [comment]
the ADL5536ARKZ are out of stock in Farnell :(

**Comment by hartytp** [comment]
Good. Remember zotino? How many times did we look at that data sheet and miss the ep connection to vss? SOmetimes a fresh pair of eyes is needed

**Comment by gkasprow** [comment]
Yes, I remember. I will pass by Technosystem tomorrow morning, maybe they have some on stock.

**Comment by hartytp** [comment]
Great, thanks!

**Comment by gkasprow** [comment]
The booster survived whole night running with full power. 
I have new ADL amplifiers so will replace it, add diode and check how it behaves with pulse input signal.

**Comment by hartytp** [comment]
Okay good! Also worth checking turn on/off standby (with/without signal applied) etc to see if there is any way to get a damaging transient

**Comment by gkasprow** [comment]
I will do it in the evening.

**Comment by hartytp** [comment]
:(

**Comment by gkasprow** [comment]
I have to leave for the meeting. Hope to be back at 5pm

**Comment by gkasprow** [comment]
full startup sequence with RF signal
![tek00025](https://user-images.githubusercontent.com/4325054/70231038-dae21800-1759-11ea-9fa5-cab048131f86.png)

**Comment by gkasprow** [comment]
The RF signal is on before the power stage supply

**Comment by gkasprow** [comment]
the off sequence
![tek00026](https://user-images.githubusercontent.com/4325054/70231381-84c1a480-175a-11ea-8f0b-866a14c5e64c.png)

**Comment by gkasprow** [comment]
Booster power on
![tek00027](https://user-images.githubusercontent.com/4325054/70231703-221cd880-175b-11ea-8fa0-8d73781692c8.png)

**Comment by gkasprow** [comment]
Brutal power off while working
![tek00028](https://user-images.githubusercontent.com/4325054/70231730-31038b00-175b-11ea-8c3b-b5245a6ce983.png)

**Comment by gkasprow** [comment]
Even during brutal power off, the correct off sequence is preserved.

**Comment by gkasprow** [comment]
The chip that was dead yesterday, connected to the 5V lab bench supply seems to be alive. It consumes 130mA of power and has 2.7V bias voltage.
It looks like the soldering process took him back to life.

**Comment by gkasprow** [comment]
Another one that was changing the gain itself consumes 110mA of power and also has 2.7V input bias voltage

**Comment by hartytp** [comment]
Interesting! Maybe it’s some other problem like a short on the board or a failure of another component

**Comment by gkasprow** [comment]
I replaced the amp with a brand new one. Before powering it on, I added 12V Zener in series with [LL4148](https://www.vishay.com/docs/85557/ll4148.pdf).
Activation of the interlock:
![tek00031](https://user-images.githubusercontent.com/4325054/70233580-2ba83f80-175f-11ea-8d2c-5238c893af5d.png)
The oscillations come from FM transmitter nearby.

**Comment by gkasprow** [comment]
This is interesting. With the clipping circuit, it achieves 11V. But this is with LL4147 capacitance and the probe capacitance.
![tek00033](https://user-images.githubusercontent.com/4325054/70233681-57c3c080-175f-11ea-877f-c0f60c75ff32.png)

**Comment by gkasprow** [comment]
more detailed view
![tek00036](https://user-images.githubusercontent.com/4325054/70234053-0f58d280-1760-11ea-8c2c-a7745ed22b3d.png)
I added traces
CH1 - ADL output
CH2 - RF switch control signal
CH3 - gate voltage
CH4 - RF output

**Comment by hartytp** [comment]
@gkasprow thanks for posting that data.

In the first plot: https://github.com/sinara-hw/Booster/issues/329#issuecomment-562088019
- I assume the blue curve (PA gate voltage) has inversion on the scope? Otherwise the bias looks positive!
- So the RF reaches the power FET after the gate is biassed but before the drain voltage is applied. That can't cause problems can it?

**Comment by hartytp** [comment]
I don't understand the second plot https://github.com/sinara-hw/Booster/issues/329#issuecomment-562089490 (off sequence)
- why is there a shift in the gate DC level when the RF stops?
- why is the gate voltage decreased before the 29V current is disabled? Couldn't that cause a very rapid discharge of the capacitors through the FET leading to damage?

**Comment by hartytp** [comment]
i.e. can you have a look at the sense resistors and see if there are current spikes during turn on/off?

**Comment by hartytp** [comment]
> I assume the blue curve (PA gate voltage) has inversion on the scope? Otherwise the bias looks positive!

Aah, no! I'm being silly. There is no inversion; when the unit is off the gate voltage is decreased to the max negative value (~-5V) to pinch off the drain current.

**Comment by gkasprow** [comment]
Exactly

**Comment by gkasprow** [comment]
Dead chips tha come back to life after heating are not a new thing. Some kind of damages in silicon can partially be reverted thanks to annealing process.

**Comment by hartytp** [comment]
I would have thought that gently removing an IC wouldn't get it hot enough to anneal the silicon?

**Comment by hartytp** [comment]
Anyway, even if that's so, it still doesn't explain how the IC was damaged.

I still wonder if the ICs are actually fine and it's something else on Booster that's the problem.

**Comment by gkasprow** [comment]
Well. I observed lack of gain and wrong dc signal. Supply was ok. DC drop on coil was 22mV.

**Comment by gkasprow** [comment]
And the fact is that with correct sine at the input the output was jumping between strange states

**Comment by gkasprow** [comment]
I did not check if the IC works. It has correct input DC only

**Comment by gkasprow** [comment]
My conclusion from the measurements above is that the new IC generates higher pulses than the old one. And it seems that the pulses are limited by the diodes. This could mean that original pulses cold be much higher possibly damaging the output stage

**Comment by gkasprow** [comment]
The IC is able to deliver up to 18dBm of power. This is 2.5Vp of sine signal. Its supply is up to 6V. So I would not be surprised if the max safe output voltage was just 8.5V. The plot of current vs output power in DS looks interesting.

**Comment by hartytp** [comment]
Ok @gkasprow I think I've lost track of all the measurements. Let me see if I understand you correctly:

- [here](https://github.com/sinara-hw/Booster/issues/329#issuecomment-562099942) you used a new (ADL5536 ) amp with no TVS. You measured the glitch when the interlock trips by looking (yellow curve) at the ADL output. The positive glitch was only ~3V on top of the DC level of 4.5V (?). So, approx 8V max. That's outside the max value but probably not destructive.
- Then you added the TVS [here](https://github.com/sinara-hw/Booster/issues/329#issuecomment-562100466). That made the glitch much bigger. Maybe 7V 0-pk. So the peak voltage is not more like 11V. That probably is enough to cause breakdown.

So, what I don't understand is:
- why does the TVS make the glitch at the amp output so much worse
- without the TVS, the glitch doesn't seem large enough to cause damage
- what about the PA stage? What's the glitch at the gate? Is it enough to cause damage to the power FET? We know what those are dying as well, is the cause the same or different?

**Comment by hartytp** [comment]
Looking at [this](https://user-images.githubusercontent.com/4325054/70234053-0f58d280-1760-11ea-8c2c-a7745ed22b3d.png) plot, it seems that the glitch at the PA gate (purple curve) is only about 3V. That's not enough to damage the power FET is it? Unless you think the change in gate voltage is causing a destructive current pulse?

**Comment by hartytp** [comment]
Basically, I'm still not convinced that the evidence we have shows a mechanism that could plausibly be leading to damage.

**Comment by gkasprow** [comment]
@hartytp let me explain.
I added the clipping circuit which clips at roughly 12V, depending on the pulse energy. It adds capacitance to the circuit of course.
Then I measured the glitch caused by the input switch. It is 3V.
And then I applied 10Hz 1.5Vpp input signal with sharp edges. This causes spikes because the first stage gets saturated. And I observe signal which is close to the clipping value and potentially dangerous to the amplifier.
So this is consistent.
I can try to remove the clipping circuit, but I think it's better to lower the clipping value to let's say 7V.
The thing that worries me the most is strange shape of the output signal that can potentially cause harm to the transistor.

**Comment by gkasprow** [comment]
Look at the gate pulse. It reaches 2V, but with an oscilloscope probe connected. It is close to the limit.
I want to add TVS diodes at the output of the first stage and second stage

**Comment by hartytp** [comment]
@gkasprow The one thing that would *really* convince me would be a measurement of a Booster with all 3 amp chips replaced with fresh ones that shows the input/outputs of the first and last stages with clear evidence of glitches that are large enough to cause damage.

I think we should do that test. Once we know that we have seen something that can cause damage, then we can add the TVS.

Related: https://www.minicircuits.com/pages/pdfs/an60034.pdf

**Comment by hartytp** [comment]
> Look at the gate pulse. It reaches 2V, but with an oscilloscope probe connected. It is close to the limit.

What's the max voltage you measured on the gate? Isn't it still within the max ratings? Isn't the probe capacitance small compared with the other capacitances around, so it can be neglected?

**Comment by gkasprow** [comment]
It's not only probe capacitance but also its bandwidth. It does not pass short pulses correctly. The amplifier stage BW is much higher than probe BW so we do not see the max amplitude correctly.
The voltage on the gate is 2V, max limit is 3V, but I'm not sure if in other channels with lower bias we don't exceed the max value.

**Comment by gkasprow** [comment]
I will replace the remaining 2 amps with the new ones in a moment. I won't remove the Zener diode for the moment.

**Comment by hartytp** [comment]
@gkasprow let me explain where I’m coming from...

Once we decide on a fix we will need to send all ~50 booster channels we have back to have the damaged amps replaced and to implement the fix. This is hugely disruptive to our experiments, time consuming and expensive.

We have been through several iterations of finding issues that look nasty but probably weren’t causing damage. So the “fixes” didn’t help.

I need to make sure that this time we are certain we’ve identified the real issue so we don’t have more international holidays for or boosters

**Comment by hartytp** [comment]
When the gate voltage gets to +2V what happens to the drain current? Does the inductor keep that low or do the caps discharge through the fet causing damage?

**Comment by gkasprow** [comment]
at 2V the drain current drops to zero. The 1uH inductor forces the current to flow and it generates a high voltage spike at the output. The spike amplitude depends on the load. In the first stage, the load is high pass filter so the load is connected only for high frequency. The second stage and the power stage have correct loads so nothing happens.
So the first stage damages due to the inductor. For the same reason, you always add a diode in parallel with relay or motor. Otherwise, the transistor will fail shortly.
My colleague who designed this amplifier assumed linear operation of the whole amplification chain.
In reality, we have moments when the amp operates in the non-linear region.
The output stage is protected from too high output voltage since it has 100V max Vds.
But its input is not protected at all!
We have IC10, loaded with 1uH coil and some weak feedback which is connected directly to the final stage via capacitors. The only circuit that prevents the second stage exceeding max gate voltage is the feedback loop of the PA stage.

**Comment by gkasprow** [comment]
Such circuit behavior would explain gain changes that I observed already and also gate threshold changes.

**Comment by hartytp** [comment]
That would explain it. I just want to see it happen on a scope or at least a simulation!

**Comment by gkasprow** [comment]
I posted a simulation where a similar RC circuit generates over 20V transients. Unfortunately, ADI do not publish SPICE models but only S-parameters.

**Comment by gkasprow** [comment]
Let's do an experiment. I will get data from the 1-st stage before, after removing the protecting diodes and again, after connecting the diodes.

**Comment by gkasprow** [comment]
before removing diodes
![tek00034](https://user-images.githubusercontent.com/4325054/70265345-76937880-179a-11ea-8262-d7f1431fcf3c.png)
after removing diodes
![tek00039](https://user-images.githubusercontent.com/4325054/70265382-8c08a280-179a-11ea-8356-bc2026f747ca.png)
diodes connected back
![tek00041](https://user-images.githubusercontent.com/4325054/70265400-975bce00-179a-11ea-99aa-6c2394f76126.png)

**Comment by gkasprow** [comment]
Look how the output shape changed after the operation

**Comment by gkasprow** [comment]
I had to change the scope scale to fit it.
It is the same circuit. I only let it run for a while with a square wave signal connected to the input without protection diode.

**Comment by gkasprow** [comment]
sorry, I mixed up screenshots. Dates do not match.

**Comment by gkasprow** [comment]
fixed. I also let it run without scope probe connected

**Comment by gkasprow** [comment]
Actually, it could be a connected probe that limits the peak voltage, not the diode. When I disconnected the probe and diode, nothing protected the 1-st stage output.

**Comment by gkasprow** [comment]
I will repeat the process and check how AC gain changes.
Steps.
- solder new IC with protection diode.
- measure AC gain
- remove the diode
- apply square wave signal
- measure AC gain
- connect the diode
- measure AC gain.

**Comment by hartytp** [comment]
That sounds prefect. If you can do that with an input signal that mimics the measured switch transients and demonstrate damage as soon as the diode is removed then I’ll be happy :)

**Comment by hartytp** [comment]
> I posted a simulation where a similar RC circuit generates over 20V transients. Unfortunately, ADI do not publish SPICE models but only S-parameters.

Sorry, I'd forgotten that. So the simulation shows that the input stage could well be damaged by the switch transients. However, if I understand correctly, it shows that the output should not be damaged, correct?

**Comment by gkasprow** [comment]
1. 1-st and 2-nd stage replaced. Input level is -10dBm. Output level is 510mV RMS
2. diode removed, Output level is 520mV RMS
3. Applied 1MHz, square wave signal, 1.7Vpp. This is the max level that does not activate the interlock
4. Applied full AC power
5. Output level is 520mV RMS
6. Diode added 529mV RMS
So, this time nothing failed.

**Comment by gkasprow** [comment]
Yes, but I didn't simulate the second stage without the load. I will do it.

**Comment by gkasprow** [comment]
It's also possible that some ICs are more fragile than the others so some channels will fail, some not.

**Comment by gkasprow** [comment]
Anyway, to give a conclusion, I would have to repeat the process above multiple times. I have to order more ICs.

**Comment by hartytp** [comment]
Well, what do we do? I would love a test that shows a clear sign of a damaging voltage at the amplifiers. If we really can't achieve that, I guess all we can do is refurbish some units (replace all amplifiers and add TVS) and then test very carefully...

**Comment by gkasprow** [comment]
I will order a few sets of ICs,  assemble them in the channels we have with TVs added. Then run some tests, remove diodes, run tests again, add diodes and see if one of them dies. Statistics should be better and hopefully, give us some solid proof. Anyway, the experiment above with altered output shape after disconnecting the diode gives some hope that this can be a solution.

**Comment by hartytp** [comment]
Okay that sounds good. Do you have an scpi synth in your lab? I can set up a script to use the synth to trigger and reset the interlock many thousands of times (this is what my fuzzer does).

**Comment by hartytp** [comment]
@gkasprow when do you expect to be able to perform those tests? Can we aim to do this next week?

**Comment by hartytp** [comment]
@gkasprow have you read that MCL app note I linked to?

I don't understand exactly the construction of the ADL amps (how is the current source constructed?), but I guess the same kinds of rules apply. In particular, having a small output capacitor with a large input capacitor means that the base bias can switch on late causing breakdown.

**Comment by gkasprow** [comment]
power on is one thing, but I did not observe peaks. We have large caps which take time to charge them.
The real issue that leads to the same damage is input signal transients

**Comment by gkasprow** [comment]
Yes, I did. It's interesting app note.

**Comment by hartytp** [comment]
Yes, it's a bit different in this case. Still I found it interesting to see how strongly exponential the process is. Small changes really can kill the amp quickly.

**Comment by gkasprow** [comment]
Yes
I have something like 12 channels in my lab.
I can replace all RF ICs, then run the tests and see which ones degrade.
I'm not sure if I should replace all the ICs. So far I noticed the damage only to the 1-st stage. So maybe we should focus on that to isolate the issue. If I replace each time 3 chips I won't know for sure which one fails.
What I can do is to measure the RF power on every stage. This would help with tracking of the faulty stage. I should also precisely measure the gate voltage and drain current.
I've just ordered 15sets of all RF ICs.
So, the procedure would look like this:
1. replace all ICs. Add TVS on 1-st and maybe 2-nd stage
2. measure RF power in three places: 1-st, 2-nd and PA stage
3. apply the square wave and record the plots as usual.
4. measure DC levels (1st stage current, PA stage current, gate voltage)
5. measure RF power in three places: 1-st, 2-nd and PA stage
6. remove TVS
7. measure RF power in 3 places
8. apply the square wave and record the plots as usual
9. measure DC levels
10. measure RF power in 3 places
11. add TVS
12- measure RF power in 3 places

**Comment by hartytp** [comment]
> If I replace each time 3 chips I won't know for sure which one fails.

My guess is that the same transient that kills the first stage kills all three. So, a transient at the first stage kills the final stage as well.

**Comment by hartytp** [comment]
> apply the square wave and record the plots as usual.

Can you make a square wave that mimmics the switch transient?

**Comment by gkasprow** [comment]
sure, just a matter of the amplitude.

**Comment by hartytp** [comment]
@gkasprow that sounds like a great plan! If the amp dies as soon as the TVS is removed then we can be pretty confident that we understand what's going on.

AFAICT the power supply rise time is very slow compared with the amp bandwidth so the issues in that app note won't be a problem.

> sure, just a matter of the amplitude.

Yes, but 100mV can make the difference between life and death so you have to get right. Maybe better to use the synth to apply CW RF power that will trip the interlock and have a script that continuously resets the interlock as fast as possible (just run the interlock reset command in a loop). Give it a few minutes to get statistics...

**Comment by hartytp** [comment]
Do you think you can look at this tomorrow? Would be great to go into the weekend knowing this is fixed :)

**Comment by gkasprow** [comment]
I don't have fresh ICs. I took a few last pieces from Technosystem and used them all today.
In Farnell, they don't have them either. So, I ordered them from Mouser.
I know what amplitude the switch causes. But I'd like to stress-test it in extremal conditions.
So instead of having 3V overshoot,I'd like to have 6V one to make sure we clearly see the effect.
We can also try it first with just switch transient, then measure gain and DC stuff. Later on, run higher pulses and repeat the process.

**Comment by hartytp** [comment]
Okay, that sounds good. Let's start with just the normal switch transients and check if that breaks it.

Hopefully the new amps arrive next week and we can finish this investigation then,

Thanks again for all the work on this.

**Comment by hartytp** [comment]
but would be good to look at the temperature readings while we wait as that's the only issue that doesn't seem related to this.

**Comment by gkasprow** [comment]
updated plan

   1. replace all ICs. Add TVS on 1-st and maybe 2-nd stage
   2. measure RF power in three places: 1-st, 2-nd and PA stage
   3. apply small square wave and record the plots as usual.
   4. measure DC levels (1st stage current, PA stage current, gate voltage)
   5. measure RF power in three places: 1-st, 2-nd and PA stage
   6. remove TVS
   7. measure RF power in 3 places
   8. apply the small square wave and record the plots as usual
   9. measure DC levels
   10. measure RF power in 3 places
   11. apply the large square wave and record the plots as usual
   12. measure DC levels
   13. measure RF power in 3 places
   14. add TVS
   15. measure DC levels
   16. measure RF power in 3 places

It's going to be a lot of work. If we discover more than let's say 2 cases, that would be enough proof and won't be necessary to destroy more ICs.

**Comment by hartytp** [comment]
If you can show that it works fine with the TVs but the switch transients kill one channel then I’m happy.

**Comment by hartytp** [comment]
@gkasprow there is one other thing we should look at during these tests.

This is a log of the 29V and 6V currents on all channels of one of our Booster. See how some channels (e.g. 1, 2, 4, 7) have basically no noise on the 29V current, but the others have quite large (few mA) glitches.

This could be an I2C/ADC programming issue, but I suspect it's real and a symptom of damage to the amp chips.

It would be really helpful if you could run a similar log for an hour or so. Identify a faulty channel then probe the sense resistor to check this is a real issue with the bias current. If that's the case, we can also identify whether the TVs help by taking an hour or so of data with/without the TVs and looking for changes. I'm happy to send you a python script that gathers the data from the logs.

**Comment by hartytp** [comment]
6V currents 

![image](https://user-images.githubusercontent.com/21218399/70312504-aaf34d00-180b-11ea-8d39-f692d4a0ff85.png)

**Comment by hartytp** [comment]
29V PA bias current:

![image](https://user-images.githubusercontent.com/21218399/70312604-dd9d4580-180b-11ea-8f76-e1213a1b541d.png)

**Comment by jordens** [comment]
They are symmetric and the same magnitude across the affected channels. Isn't that weird if avalanche or EOS damage is the hypothesis? Smells digital. What does the sample histogram look like?

**Comment by hartytp** [comment]
> What does the sample histogram look like?

![image](https://user-images.githubusercontent.com/21218399/70315263-1c81ca00-1811-11ea-9122-6b6dc77354d8.png)

**Comment by gkasprow** [comment]
Ok, I will measure with a multimeter and log the current. I'm just organizing a person who will help mi with documentation, logging and soldering. We will start once components arrive.

**Comment by gkasprow** [comment]
To make life easier we can make voltage and current logging using second 4-channel scope.

**Comment by hartytp** [comment]
I stuck a probe on after the InAmp, but I'm seeing a lot of SMPS pick up. Probably the ground point isn't good enough. It's not totally straightforward to see this since we're only looking at 10mV or so.

I don't have a good differential probe. Would you mind having a go at this next week @gkasprow? The main thing is to determine whether this is another I2C issue or something real on the bias current.

**Comment by gkasprow** [comment]
I'm just preparing with one of colleagues from Creotech to the debugging process.
We will connect two scopes with probes to various signals in the Booster. Then we will use Python scripts to read out and log real data including screenshots and DC levels. I have also multimeters that can be accessed from Python, so we will measure the smallest current drifts. The components from Mouser were [already shipped](https://www.fedex.com/apps/fedextrack/?action=track&tracknumbers=136473124515)

**Comment by hartytp** [comment]
Awesome!

**Comment by gkasprow** [comment]
The components arrived. Kuba helped me with the scripting of the measurement process. So, tomorrow we will be fighting.

**Comment by hartytp** [comment]
That's great!

**Comment by hartytp** [comment]
hopefully we can end this saga this week :)

**Comment by hartytp** [comment]
@gkasprow how did you get on with this?

**Comment by gkasprow** [comment]
OK, so it was a long, exhausting day full of work.
We disassembled all 10 plugin units, removed all amplifiers and started adding TVS diodes. It is laborious work because in certain places one must remove solder resist to get access to the copper. The TVS I ordered are very tiny so each must be soldered under a microscope. It takes a lot of time.
 I noticed after the removal of the power stages two of them were not wetted by tin. Two were wetted partially, less than 25%. This causes poor thermal contact and certainly leads to rapid overheating and quick degradation of parameters. This would explain some issues.
It looks like some transistors were kept too long and oxidized prior assembly. One must simply put such transistors covered by activating fulx to the oven for 24h. This is the process we have to implement for this particular component. Such a situation cannot be easily diagnosed by AOI and the only method that gives 100%  chance is either pre-tinning of the thermal pad or later diagnostic using a thermal camera. I will discuss the issue with production people to make sure it would not happen anymore.
I got a set of an early version of preamp boards and we desoldered the PHA amplifiers. But unfortunately, my colleague who was assisting me mixed the ones I desoldered with the ones he removed. I have forgotten to tell him which box is which. There is no way how we can distinguish them. I managed to patch only 3 boards with new amps and TVS diodes. So these can be tested.
So, to continue and get better statistics, I have to order new amps from MiniCircuits to keep the experiment conditions clean.
In the meantime, we built the test suite. So with simple scripts, we log all RF signals in the chain, output powers, biases, voltages. Bias value is measured with DMM, rest with high-end scopes. Moreover, we log also the booster diagnostics.
In this way, after we change anything, with one click will dump all results in a dedicated directory
Tomorrow I leave for the conference in Crakow and will continue on Monday. I thought we manage today but mechanical disassembly and TVS insertion took far much longer than I expected.
A few photos. They close University at 9pm so I had to leave. We are very close!
![2019-12-12 19 54 43](https://user-images.githubusercontent.com/4325054/70745334-d8685b00-1d23-11ea-881b-61cb73b3af33.jpg)
![2019-12-12 19 54 46](https://user-images.githubusercontent.com/4325054/70745345-dd2d0f00-1d23-11ea-9336-ae07844adbee.jpg)
![2019-12-12 19 54 49](https://user-images.githubusercontent.com/4325054/70745377-ed44ee80-1d23-11ea-85e2-5cffae28f88e.jpg)

**Comment by hartytp** [comment]
Good work! These things take quite some love to debug but we’re getting there!

Hopefully Monday will be our day. No more booster bugs would be a good Christmas present ;)

**Comment by gkasprow** [comment]
I"m testing the power block with TVS added. And one module has very low gain. Another one has 5V overload. These TVS which are specified for 6V2 operation for som reason short the power rail.
I measured one of them and it has 4.5V at 20mA current so there is something wrong with package content.

**Comment by gkasprow** [comment]
luckily I ordered bidir version of it which has 8V at 20mA. But I have to remove the old one and install new ones...

**Comment by gkasprow** [comment]
Now it works perfectly fine with TVS diodes.

**Comment by hartytp** [comment]
> luckily I ordered bidir version of it which has 8V at 20mA. But I have to remove the old one and install new ones...

That's annoying! Glad the other ones work.

Does that mean you now have all the parts and can test tomorrow?

**Comment by gkasprow** [comment]
I'm testing it right now. Surprisingly one of the modules had 3V3MP shorted to GND during PCB production. I have no idea how it could work before. It looks like one has to change the PCB provider because the current one does not provide electrical tests even though it charges for such.
I'm logging the gate voltage with high precision DMM and also all signals and currents.

Pin=-10dBm, Out goes via 20dB attenuator to the scope. The first measurement was taken just after power on.

|   time   |     Vg     |    PWR[Vrms]      |  Icc [A] | temp |
| -------- | ---------- | ----------------- | ---- | ---- |
| 19:23  |  -1.16893673  | 0.1550506 | 0.113 | 28
| 21:04| -1.1688882 |  0.1540794 | 0.111 | 28 |

**Comment by gkasprow** [comment]
I will dump results every hour untill late night. Tomorrow I will remove the TVS and follow the test protocol  I wrote above.

**Comment by gkasprow** [comment]
Hopefully, I will get new ICs from minicircuits and will assemble the remaining 8 boards.

**Comment by hartytp** [comment]
> Surprisingly one of the modules had 3V3MP shorted to GND during PCB production. I have no idea how it could work before. It looks like one has to change the PCB provider because the current one does not provide electrical tests even though it charges for such.

Odd! I think we've tested all 8 channels so I suspect this worked at one point.

Do you think it could be another mechanical issue like before?

**Comment by hartytp** [comment]
> I'm logging the gate voltage with high precision DMM and also all signals and currents.

:+1: good idea. That will help to get nice reliable diagnostics and catch this damn bug red handed.

Is this with CW RF in, or are you applying your square wave to simulate the switch glitches that caused damge?

> I will dump results every hour untill late night. Tomorrow I will remove the TVS and follow the test protocol I wrote above.

> Hopefully, I will get new ICs from minicircuits and will assemble the remaining 8 boards.


Perfect!

**Comment by gkasprow** [comment]
I'm testing it with a low-level signal right now. Want to observe the stability of the gate voltage and current. After I do stress-tests without TVS, I will do the same long term tests.

**Comment by gkasprow** [comment]
It's hard to say. I discover new issues every time I touch it :)

**Comment by hartytp** [comment]
> It's hard to say. I discover new issues every time I touch it :)

Yes...sigh...

Well, it's easy to check just see if there is still a short on the board when it's removed from the module. Last time the short wen away when the screws were even loosened. Basically, someone needs to take more care: the PCB house or the machine shop, but it's worth knowing who to yell at ;)

**Comment by hartytp** [comment]
> I'm testing it with a low-level signal right now. Want to observe the stability of the gate voltage and current. After I do stress-tests without TVS, I will do the same long term tests.

Good idea. So the idea is:
1. Log with small signal applied
2. Stress test with TVS
3. Log with small signal
4. Stress test without TVS
5. Log

That sounds like a great plan as we will be sure why these boards died

**Comment by hartytp** [comment]
aah, yes, this was the plan. That's a great idea https://github.com/sinara-hw/Booster/issues/329#issuecomment-562352523

**Comment by hartytp** [comment]
How did this go overnight?

**Comment by gkasprow** [comment]
The current seems to be dropping a little bit, but I measure it with scope, after the amplifier, so could be day-time effect.

**Comment by gkasprow** [comment]
It's time to finally treat it with pulses.

**Comment by gkasprow** [comment]
I treated it with 1.7V 100kHz square wave. 1.8V caused protection activation.

**Comment by gkasprow** [comment]
after 30mins of pulses no gain changes were observed

**Comment by gkasprow** [comment]
Let's remove all TVS then

**Comment by gkasprow** [comment]
There is scope influence as well. Let's log the parameters without scope as well.

**Comment by gkasprow** [comment]
Probes capacitance may influence the rise time and transistor breakdown

**Comment by gkasprow** [comment]
It works without TVS, the gain is slightly higher since TVS have non-negligible capacitance.
Let's try it with pulses and see what happens.

**Comment by gkasprow** [comment]
I left probes connected for the moment.

**Comment by hartytp** [comment]
Okay moment of truth

**Comment by gkasprow** [comment]
I observed a slight gain increase. Let's remove the probes and try it again with pulses.

**Comment by gkasprow** [comment]
I assembled the TVS back and will leave it overnight logging the results.

**Comment by hartytp** [comment]
Great. Did you notice any obvious side effects from hammering it with a square wave without TVS?

**Comment by gkasprow** [comment]
No clear conclusion. Let's leave it overnight working and tomorrow repeat with other modules.

**Comment by gkasprow** [comment]
Here are the [results](https://github.com/sinara-hw/Booster/issues/329#issuecomment-566223316)

**Comment by gkasprow** [comment]
Some variations just after HW changes could be caused by board heating due to soldering.

**Comment by gkasprow** [comment]
I still didn't receive the package from Minicircuits.

**Comment by hartytp** [comment]
> The current seems to be dropping a little bit, but I measure it with scope, after the amplifier, so could be day-time effect.

Interesting. The amps have a negative temp co, so could be that? Can I see the data?

**Comment by hartytp** [comment]
> No clear conclusion. Let's leave it overnight working and tomorrow repeat with other modules.

:+1: If we can see a case where removing the TVS and applying the square waves causes a unit to break promptly then I'll be very happy :)

**Comment by gkasprow** [comment]
All the data  are here:
[Measurements.zip](https://github.com/sinara-hw/Booster/files/3975166/Measurements.zip)

**Comment by gkasprow** [comment]
That's why I want to repeat a shorter version of the procedure above with at least 10 modules. So far I managed to break only one IC in such a way.

**Comment by hartytp** [comment]
That sounds like a very good idea

**Comment by gkasprow** [comment]
nothing changed during the night. Let's switch to another module.

**Comment by hartytp** [comment]
:( always the way...things only break when you don’t want them to. Did you use a large enough input square wave? It’s very non linear so the input must be big enough to cause damage

**Comment by gkasprow** [comment]
I use 1.7Vpp. The input stage is overdriven because I observe large peak at its output. Higher voltage causes protection activation. I can also use very short negative pulses. I've just applied -3V 18ns pulses and it does not activate the interlock. Let's use them.

**Comment by gkasprow** [comment]
So, I took another module. I connected TTL signal which gives sharp ns edges
TVS assembled, probes connected, Icc calibrated 50mA
11.07 RF, -10dBm, out = 161mV, 
11.09  100kHz,  sharp TTL pulses for 10mins
11.27 Ic=45mA, out voltage dropped to 125mV
11.28 bias calibrated to 50mA. Before was 1249, now is 1229
11.31 output is still 125mV

**Comment by hartytp** [comment]
Okay that’s good! So the conclusion is that sharp rise time glitches can cause reduction in the bias current and gain changes. So that does explain what we have seen.

However are you saying that the damage still occurred with the TVS fitted?

**Comment by gkasprow** [comment]
I removed all TVS
11.39 Out voltage is 140mV. Ibias is 50mA
11.40. I'm applying TTL pulses. Before, with TVS, the 1st stage voltage was clipped to 8V. Now is 12V.
11.45. Out voltage is 137mV. Ibias is 50mA

**Comment by gkasprow** [comment]
the TVS protect the output stages. I didn't add TVS that protects FET input.

**Comment by gkasprow** [comment]
Let's do the following:
- replace the JFET and repeat.
- try the same with unit I tested yesterday.

**Comment by gkasprow** [comment]
I repeated it with second channel.
TVS are assembled. No probes connected
13:20. -10dBm,Vout=147mV
13:23. TTL pulses for 5min
13:28 -10dBm, Vout = 145mV

**Comment by gkasprow** [comment]
I removed the TVS. No probes
13:37. -10dBm, Vout = 158mV
13.39 TTL pulses
13:53 -10dBm, Vout = 155mV

**Comment by gkasprow** [comment]
It looks like this unit is resistant to transients

**Comment by gkasprow** [comment]
I replaced the JFET in previous module no 8
14:33 -10dBm, Vout = 127mV
14.34 TTL for 5 mins

**Comment by gkasprow** [comment]
The first stage has lower gain than before. Let's replace it.

**Comment by gkasprow** [comment]
I have a theory that the input stage of the ADL chip breaks down, not its output.

**Comment by gkasprow** [comment]
FET breaks as well, but it is other story.

**Comment by gkasprow** [comment]
After the replacement of the first and third stage, the output voltage is 130mV

**Comment by gkasprow** [comment]
15.44 replaced 1-st stage in module 8. TTL pulses for 5 mins
15:51 -10dBm,  output is 124mV

**Comment by gkasprow** [comment]
Icc = 46mA, after calibration is 1298, before it was 1309

**Comment by gkasprow** [comment]
16:05 TVS removed, Vout = 145mV
16:06 TTL pulses
16:15 -10dBm , Vout = 138mV
18:49 -10dBm, Vout = 140mV

**Comment by gkasprow** [comment]
So, the conclusion is that using the channel with or without TVS causes gain changes and output bias changes.
Let's add TVS at the very input of the channel. And TVS after 1-st stage and at the input of the 3-rd stage. And of course let's replace the amps.

**Comment by gkasprow** [comment]
the difference between channels 8 and 10 is such, that channel 8 has high-Q inductors. Channel 10 has lower Q inductors.

**Comment by gkasprow** [comment]
I added TVS at the input and also 4V TVS at the FET gate.
Let's see if I can damage the amp!

**Comment by gkasprow** [comment]
Module 8
19:51 -10dB Vout = 124mV
19:53 TTL
19:54 connected probes, acquired signals. They are seriously clipped
19:55 as above, other polarity
19:59

**Comment by gkasprow** [comment]
It looks like this time I didn't manage to break the amplifier. The gain is slightly lower due to four TVS capacitances. We can choose ones with lower capacitances. They come in 0201 packages so I had to use bigger ones with higher C

**Comment by gkasprow** [comment]
@hartytp the conclusion is that today I managed to break one amp seriously and another amp slightly even with TVS after 1-st, 2-nd and 3-rd stage.
When I added TVS at the very input and at the gate, I didn't manage to harm the amplifier.
My recommendation is to use lower capacitance TVS at the very input of the amp, after first stage and at the FET gate.

**Comment by gkasprow** [comment]
I'm just modyfiying 3 channels (I add 3 TVS only) which I will treat with TTL pulses

**Comment by gkasprow** [comment]
channel 7, 3xTVS added, no output TVS
20:52  -10dB, Vout=142mV, Ib=49mA,
20:54 TTL pulses for 5mins
20:59 -10dB, Vout = 142mV, Ib=46mA

**Comment by gkasprow** [comment]
channel 11 (old revision, HiQ), 4x TVS added, also at the channel output
21:32 -10dBm, Vout = 150mV
21:35 TTL pulses
21:40 -10dBm, Vout = 152mV

**Comment by gkasprow** [comment]
output TVS removed, 
21.40 -10dBm, Vout 160mV
21.41 TTL pulses
21.43 -10dBm, Vout 163mV

**Comment by gkasprow** [comment]
The effect is not very strong, but it looks like we also need the output TVs as well.
The 4V TVS at the FET input protects it against the permanent change of Vth voltage.
The TVS I use have 2.5pF so they affect the gain. I purchased also lower capacitance TVS, <0.5pF but they are too small (0201, pads below) to be added to an existing design.
Adding the TVS is easy, just solder between the RF track and GND.
The input amplifier already died or lost gain three times.
It can be killed from the input or output side, both are possible. Since we are operating with low input power, TVS at the input won't harm the signal. We must also use TVS at the output of the first stage due to the overvoltage issue. It looks like we exceed the max positive FET gate-level which causes Vth degradation. TVS shorts to GND every positive voltage above 0.7V as well as negative below 4V.

**Comment by hartytp** [comment]
@gkasprow cool!

So, summarising what I think the conclusions are:
- You've applied a square-wave input with an amplitude/rise time chosen to approximately match the transients produced by the interlock switch
- This broke several channels, both in the pre-amp stage and in the PA/FET stage. Breakage was evidenced by changes in the current draw (for a given bias voltage) and the gain
- Adding TVS at the input/output of all stages reliably prevented damage
- In separate tests, we've looked at all other potential transients (power on/off, standby/enable, etc) and not seen anything suspicious
- so these glitches can explain all symptoms we've seen and can be reliably fixed by the TVS didodes. Moreover we've done all the other tests we can think of and didn't find anything suspicious 

Is that correct? If so, I think we can close this issue!

**Comment by gkasprow** [comment]
That's correct. I will patch a few channels, test their response and send them to you.

**Comment by hartytp** [comment]
Okay, great! Can you patch (including replacing all three ams for each channel) an entire Booster chassis to send back to it.

Before returning, can you have a look at the final issues here: https://github.com/sinara-hw/Booster/issues/343

I suspect it's all SW/I2C issues but we should quickly rule out HW problems.
